common/djangoapps/student/views.py

```diff
@@ -226,37 +227,51 @@ def reverification_info(course_enrollment_pairs, user, statuses):
     return reverifications
 
 
-def get_course_enrollment_pairs(user, course_org_filter, org_filter_out_set):
+def get_course_enrollments(user, course_org_filter, org_filter_out_set):
     """
-    Get the relevant set of (Course, CourseEnrollment) pairs to be displayed on
-    a student's dashboard.
+    Given a user, return a filtered set of his or her course enrollments.
+
+    Arguments:
+        user (User): the user in question.
+        course_org_filter (str): if not None, ONLY courses of this org will
+            be returned. For use in Microsites.
+        org_filter_out_set (list[str]): if course_org_filter is None, then any
+            course whose org is in this list will NOT be returned.
+    Notes:
+        Out of course_org_filter and org_filter_out_set, exactly one should
```

Comment:
@nasthagiri I'm not clear on what would make this function in particular a good case for PyContracts if I can't make the type of one parameter depend on the type of the other... wouldn't the thing I would want to say with PyContracts be "this variable is None if and only if the other one is non-None, and vice-versa"?